test_package/conanfile.py

```diff
@@ -1,19 +1,20 @@
-import os
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
 from conans import ConanFile, CMake, tools, RunEnvironment
+import os
 
 
-class LksctpTestConan(ConanFile):
+class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
     generators = "cmake"
 
     def build(self):
         cmake = CMake(self)
-        cmake.configure(source_dir=self.conanfile_directory, build_dir=os.getcwd())
+        cmake.configure()
         cmake.build()
 
     def test(self):
         with tools.environment_append(RunEnvironment(self).vars):
-            if self.settings.os == "Windows":
-                self.run(os.path.join("bin","test_package"))
-            else:
-                self.run("DYLD_LIBRARY_PATH=%s %s"%(os.environ['DYLD_LIBRARY_PATH'],os.path.join("bin","test_package")))
+            bin_path = os.path.join("bin", "test_package")
+            self.run("LD_LIBRARY_PATH=%s %s" % (os.environ.get('LD_LIBRARY_PATH', ''), bin_path))
```

Comment:
If you want to create a test for `with_sctp` scenario, I'd suggest something like this:

```cpp
#include <sys/types.h>          /* See NOTES */
#include <sys/socket.h>
#include <netinet/in.h>
#include <netinet/tcp.h>
#include <netinet/sctp.h>

int main(int argc, char **argv) {
    int fd = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP );

    int prot;
    socklen_t len = sizeof(prot);
    getsockopt( fd, SOL_SOCKET, SO_PROTOCOL, &prot, &len );

    if (strcmp(argv[1], "with_sctp") == 0) {
        return !(prot == IPPROTO_SCTP);
    } else {
        return !(prot == IPPROTO_TCP);
    }
}
```

Then you ran it twice, once w/ "with_sctp" argument and "libwithsctp" in library path, and other without argument and without "libwithsctp" in library path.

Reply:
Really nice, I'll use this one.